### PR TITLE
CNV-33431: Storage profile must be reapplied if storage class changes

### PR DESCRIPTION
**Pull request by aspauldi**
Version(s): 4.15+

Issue: [CNV-33431](https://issues.redhat.com/browse/CNV-33431)

Links to docs previews: 
- [Customizing the storage profile](https://69785--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/storage/virt-configuring-storage-profile#virt-customizing-storage-profile_virt-configuring-storage-profile)
- [Configuring storage profiles](https://69785--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/storage/virt-configuring-storage-profile)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by aspauldi** [comment]
@aglitke, Can you take a look at the paragraph I added to Customizing the storage profile?  I wondered if the update should go in the Configuring storage profiles section instead, but the content does pertain to customizations.

**Review by aglitke — APPROVED** [review]
lgtm

**Comment by aspauldi** [comment]
@geetikakay, can you please review this PR for me? I don't see a QE person named in the Jira, but I looked at the Feature and see that you have a story for test automation. If you are not the right QE person to review, can you let me know? Thanks!

**Comment by geetikakay** [comment]
@aspauldi updates to existing sc parameters are forbidden, so this patch looks okay but since it's for storage i think @jpeimer knows better

**Comment by aspauldi** [comment]
Thanks, @geetikakay.

@jpeimer, can you please review this PR? Thanks!

**Review by jpeimer — APPROVED** [review]

File: modules/virt-customizing-storage-profile.adoc [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 You can specify default parameters by editing the `StorageProfile` object for the provisioner's storage class. These default parameters only apply to the persistent volume claim (PVC) if they are not configured in the `DataVolume` object.
 
+You cannot modify storage class parameters. To make changes, delete and recreate the storage class. You must then reapply any customizations made to the storage profile.
```
Comment (mramendi) [inline modules/virt-customizing-storage-profile.adoc:12]: Please use "re-create" not "recreate" as per IBM Style Guide https://www.ibm.com/docs/en/ibm-style?topic=word-usage

But "reapply" is correct.

I would also expand to at least "You must then reapply any customizations that were previously made to the storage profile." Arguably to "If you modified the storage profile previously, reapply the changes after re-creating the storage class"; this is better style but ignores the fact that someone else might have made the original changes. Your call.

File: virt/storage/virt-configuring-storage-profile.adoc [hunk 2]
```diff
@@ -6,7 +6,7 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-You must configure storage profiles if your storage provider is not recognized by the Containerized Data Importer (CDI)
+You must configure storage profiles if your storage provider is not recognized by the Containerized Data Importer (CDI).
```
Comment (mramendi) [inline virt/storage/virt-configuring-storage-profile.adoc:9]: I would suggest rewriting to active voice: "You must configure storage profiles if the Containerized Data Importer (CDI) does not recognize your storage provider." Or possibly even: "If the Containerized Data Importer (CDI) does not recognize your storage provider, you must configure storage profiles".

**Review by mramendi — COMMENTED** [review]
As I am in the process of induction into the peer review squad, @JoeAldinger will finish this review.

**Pull request merged**